scripts/blender_load_replay.py

```diff
@@ -0,0 +1,348 @@
+                    last_scene_idx = i
+
+            all_c = keyframe["creations"][last_scene_idx:]
+            for creation_dict in all_c:
+                filepath = creation_dict["creation"]["filepath"]
+                if "cubeSolid" in filepath:
+                    continue
+                if "bellow" in filepath:
+                    continue
+                obj = import_item(ImportItem(filepath))
+                if "scale" in creation_dict["creation"]:
+                    obj.scale = creation_dict["creation"]["scale"]
+                instance_key = creation_dict["instanceKey"]
+                render_asset_map[instance_key] = obj
+
+                filepath = filepath.split("?")[0]
+                asset_info_by_key[instance_key] = asset_info_by_filepath[
+                    filepath
+                ]
+
+        if "stateUpdates" in keyframe:
+            for update_dict in keyframe["stateUpdates"]:
+                instance_key = update_dict["instanceKey"]
+                translation = update_dict["state"]["absTransform"][
+                    "translation"
+                ]
+                rotation = update_dict["state"]["absTransform"]["rotation"]
+                if instance_key not in render_asset_map:
+                    continue
+                obj = render_asset_map[instance_key]
+
+                obj.rotation_mode = "QUATERNION"
+
+                asset_info = asset_info_by_key[instance_key]
+
+                # note coordinate convention change for Blender
+                obj.location = (
+                    translation[0],
+                    -translation[2],
+                    translation[1],
+                )
+                obj.rotation_quaternion = (
+                    rotation[0],
+                    rotation[1],
+                    -rotation[3],
+                    rotation[2],
+                )
+
+                frame = asset_info["frame"]
+                if frame["up"] == [0.0, 1.0, 0.0]:
+                    pass
+                elif frame["up"] == [0.0, 0.0, 1.0]:
+                    obj.rotation_mode = "XYZ"
+                    obj.rotation_euler[0] -= 1.5708
+                else:
+                    raise NotImplementedError(
+                        "unexpected coordinate frame " + frame
+                    )
+
+        if do_add_anim_keyframes:
+            for instance_key in render_asset_map:
+                obj = render_asset_map[instance_key]
+                obj.keyframe_insert(data_path="location", frame=keyframe_index)
+                obj.keyframe_insert(
+                    data_path="rotation_quaternion", frame=keyframe_index
+                )
+
+    for o in bpy.context.scene.objects:
+        o.hide_set(False)
+    # To fix import issue where some robot models would have transparent links.
+    for m in bpy.data.materials:
+        m.show_transparent_back = False
+
+    add_lights = settings.get("lights", [])
+    for light in add_lights:
```

Comment:
This is pretty messy. I hate introducing yet another JSON format file. @0mdc has already added light support to gfx-replay. Can we just use that? I.e. update the keyframe parsing here to parse the lights. And then your command line could parse multiple gfx-replay files. And one of them just specifies the lights.

Reply:
Btw, I find that Blender area lights produce the prettiest results, and we can't currently represent them directly in gfx-replay (it only supports point lights). Not sure what to do about that. What about loading an additional *Blender* file as a kind of template?

Again, what I really don't like is introducing yet another proprietary JSON format file.